cmd/ipfs/daemon.go

```diff
@@ -288,9 +291,30 @@ func daemonFunc(req *cmds.Request, re cmds.ResponseEmitter, env cmds.Environment
 			return fmt.Errorf("fs-repo requires migration")
 		}
 
+		// Read from existing config
+		cfg, err := cctx.GetConfig()
```

Comment:
It seems like allowing an entire config to be specified for a temp node also invites many problems, as users will not necessarily know what the new configuration looks like until after the migration and may typically just use their existing config (leading to issues mentioned above).  Also, the configuration for a temp, used only to download a migration or two, does not need more than a default config.  Users not knowing this may think it best to use a config similar to what they are using for their real node.

Instead of a config file, WDYT about letting the user specify peer(s) on the command line, if they happen to know one 
 or more that may be useful for migration?

Reply:
> It seems like allowing an entire config to be specified for a temp node also invites many problems...

Ya, it's using a pretty big hammer to deal with a smaller set of problems but needing to modify the config for the purposes of migrations at all seems like a potentially advanced requirement for an end user, perhaps sufficiently so that they might as well get access to the full range of tools.

However, there are also potentially a large number of options that could be specified during a migration so idk if command line flags for each of them is really a reasonable alternative. For example, if someone is on a private network and therefore needs the network symmetric key they're a bit out of luck with just some peer flags. If we make our routing systems more configurable we're similarly out of luck.

cc @Stebalien as this and https://github.com/ipfs/go-ipfs/pull/8064#discussion_r611867631 seem to be related to the comments in the proposal PR (https://github.com/protocol/web3-dev-team/pull/86/files#diff-991744cab1b93840a21929cbf5aa7bc00a37ae4244b5c569df086214e36e47cfR66 and https://github.com/protocol/web3-dev-team/pull/86/files#r610934803)

Reply:
> An alternative way to do this is to have a --migration-temp-node-config (or better name blush) option that takes a config file to use for the temporary node. This could protect us from a variety of bugs/corner cases as it allows users to configure the temporary node as if it was a regular one.

We _could_, maybe, later, support that for advanced users/distros. That's not going to really help in this case because users will be _very_ confused about why their migration options aren't being applied. Users are going to be less confused if we start a temporary node that ignores their config.

---

I've left a comment on the proposal. IMO, we should just try to read the relevant part of the config and ignore the rest. Downsides:

1. Migrations can't touch the migration part of the config. Honestly, I'm fine with that.
2. If we decide to move the config, things could get... tricky. But trying a set of configs in order of precedence, _just_ to read a small portion probably isn't a huge issue.

We could also do this on the command-line, but it could get icky (as @aschmahmann said, lots of options).

Reply:
I think I'm fine with this approach. If we define the migrations config options as described in the proposal and then we try to deserialize the config file (or object) into a struct that only has the migration values that should be fine.

This will mean that unlike the rest of the config which is nominatively typed in that backwards incompatible changes are tied to a repo version, that we are likely to use structural typing for the migrations specific section (although if this proves too hard we can add a field into the migrations struct to track versioning).

This reliance on structural typing means we basically don't have to worry too much about the potential migration config changes for the time being and it's possible we'll end up with a repo/package that contains all the migration configs (like what we used to do with fs-repo-migrations, but really small since it's only for a portion of the config file).

For the time being we can probably punt on dealing with private swarms or people with custom bootstrap nodes since they can:
1. Just use HTTP, like they've been doing so far
2. Download the migrations binary manually, or run the full fs-repo-migrations binary or ipfs-update

If this becomes a problem and we need the sledge hammer of passing in `--migration-temp-node-config` we can do that, but let's delay until we have a user + use case who needs this.

Reply:
I am going to leave the configuration for a later time.  Currently, go-ipfs accepts a couple of optional flags that allow the user to specify how to download the migrations (`--download-migration`) and if/how to keep downloaded migrations (`--keep-migration`).  Even if this information is available from a config file in the future, I think these flags will will still be useful to override that.

Reply:
So, I thought this solution would be fine... but I've changed my mind. We have _way_ too many special-purpose daemon flags that should be settings. We're already trying to get rid of the existing daemon flags, I don't want to add more that we'll need to maintain into the future for backwards compatibility.

If we want to be able to override config options, we need a coordinated system to do that but this isn't that.

Reply:
@Stebalien Done.  Now reading migration setting from config file, and using defaults for any values not present.  Removed CLI flags for specifying migration configuration.